### ISSUE TITLE
No accessibility heading in react-native Alert dialog

### ISSUE THREAD

**Issue by jp833** [facebook/react-native#32876]
### Description

For any dialog there should be an accessibility heading. In case of the react-native Alert dialog it should be the title of the dialog. However, on iPhone, there is no heading found. Repro steps described below.

### Version

.0.63.4

### Output of `npx react-native info`
--

### Steps to reproduce

1. On your iphone launch alert dialog with the example given in the documentation https://reactnative.dev/docs/alert. 
2. Turn on phone accessibility voice over.
3. Launch the dialog
4. Use the voice over rotor to select headings - like shown here https://support.apple.com/en-us/HT204783

Expected: It should announce the heading of the dialog which is basically the 'title'.
Actual: No voice over as no headings were found. 

### Snack, code example, screenshot, or link to a repository

Code from here should be good https://reactnative.dev/docs/alert.

**Comment by blavalla** [comment]
Thanks for the issue! This same problem is covered by [30860](https://github.com/facebook/react-native/issues/30860).
